tracer/src/Datadog.Trace/AgentProcessManager.cs

```diff
@@ -74,39 +74,30 @@ public static void Initialize()
                     return;
                 }
 
-                var automaticTraceEnabled = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.TraceEnabled, string.Empty)?.ToBoolean() ?? true;
-                var automaticProfilingEnabled = EnvironmentHelpers.GetEnvironmentVariable(ContinuousProfiler.ConfigurationKeys.ProfilingEnabled)?.ToBoolean() ?? false;
-
-                if (azureAppServiceSettings.CustomTracingEnabled || automaticTraceEnabled || automaticProfilingEnabled)
+                if (string.IsNullOrWhiteSpace(TraceAgentMetadata.ProcessPath))
                 {
-                    if (string.IsNullOrWhiteSpace(TraceAgentMetadata.ProcessPath))
-                    {
-                        Log.Warning("Requested to start the Trace Agent but the process path hasn't been supplied in environment.");
-                    }
-                    else if (!Directory.Exists(TraceAgentMetadata.DirectoryPath))
-                    {
-                        Log.Warning("Directory for trace agent does not exist: {Directory}. The process won't be started.", TraceAgentMetadata.DirectoryPath);
-                    }
-                    else
-                    {
-                        Processes.Add(TraceAgentMetadata);
-                    }
+                    Log.Warning("Requested to start the Trace Agent but the process path hasn't been supplied in environment.");
+                }
+                else if (!Directory.Exists(TraceAgentMetadata.DirectoryPath))
```

Comment:
This is not part of your changes, but shouldn't we also check if the executable file exists, not only the directory?

Reply:
Yep, makes sense, I'll add check for that 👍

Reply:
Done in `StartProcesses()`